src/rgw/rgw_sal_motr.cc

```diff
@@ -4187,15 +4183,14 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
     }
   }
 
-  string obj_part_iname = "motr.rgw.object." + tenant_bkt_name + "." + mp_obj.get_key() +
-                          "." + upload_id + ".parts";
-  ldpp_dout(dpp, 20) <<__func__ << ": object part index=" << obj_part_iname << dendl;
+  string iname = "motr.rgw.bucket." + tenant_bkt_name + ".multiparts";
+  ldpp_dout(dpp, 20) <<__func__ << ": object part index=" << iname << dendl;
   key_vec[0].clear();
-  key_vec[0] = "part.";
+  key_vec[0] = mp_obj.get_key() + "." + upload_id;
   char buf[32];
-  snprintf(buf, sizeof(buf), "%08d", marker + 1);
+  snprintf(buf, sizeof(buf), ".%08d", marker + 1);
   key_vec[0].append(buf);
-  rc = store->next_query_by_name(obj_part_iname, key_vec, val_vec);
+  rc = store->next_query_by_name(iname, key_vec, val_vec);
```

Comment:
@andriytk : As this Motr's next_KV operation can return parts of some other multipart upload (due to the fact that now parts of different multipart sessions are stored in the same index), we would need a logic in this list_parts function to discard those parts that belong to some other multipart upload (i.e. with different upload_id) and stop further enumeration, so that list_parts returns only the parts of the given upload id.

Reply:
Very true, will add it. Thanks, @DPG17!

Reply:
Fixed. @DPG17, please check.